sklearn/mixture/gmm.py

```diff
@@ -411,7 +411,30 @@ def sample(self, n_samples=1, random_state=None):
                     num_comp_in_X, random_state=random_state).T
         return X
 
-    def fit(self, X, y=None):
+    def fit_predict(self, X, y=None):
+        """
+        Fit and then predict labels for data.
+        Warning: due to the final maximization step in the EM algorithm,
+        with low iterations the prediction may not be 100% accurate
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+
+        Returns
+        -------
+        C : array, shape = (n_samples,)
+        """
+
+        responsibilities = self._fit(X, y)
+        if responsibilities is None:
+            prediction = None
```

Comment:
I would return directly:

```
return self._fit(X, y).argmax(axis=1)
```

and make sure that `_fit` can never return `None` (make it raise a `ValueError` or similar with a meaningful error message otherwise).